--- a/configure.ac
+++ b/configure.ac
@@ -4348,17 +4348,22 @@
         fi
         # Set up other core UNIX files
         SOURCES="$SOURCES $srcdir/src/core/unix/*.c"
-        # The iOS platform requires special setup.
-        AC_DEFINE(SDL_VIDEO_DRIVER_UIKIT, 1, [ ])
-        if test x$enable_video_opengles = xyes; then
-            AC_DEFINE(SDL_VIDEO_OPENGL_ES2, 1, [ ])
-            AC_DEFINE(SDL_VIDEO_OPENGL_ES, 1, [ ])
-            AC_DEFINE(SDL_VIDEO_RENDER_OGL_ES, 1, [ ])
-            AC_DEFINE(SDL_VIDEO_RENDER_OGL_ES2, 1, [ ])
-            EXTRA_CFLAGS="$EXTRA_CFLAGS -DGLES_SILENCE_DEPRECATION"
+        AC_ARG_ENABLE(video-uikit,
+[AS_HELP_STRING([--enable-video-uikit], [use UIKit video driver [default=yes]])],
+                  , enable_video_uikit=yes)
+        if test x$enable_video = xyes -a x$enable_video_uikit = xyes; then
+            # The iOS platform requires special setup.
+            AC_DEFINE(SDL_VIDEO_DRIVER_UIKIT, 1, [ ])
+            if test x$enable_video_opengles = xyes; then
+                AC_DEFINE(SDL_VIDEO_OPENGL_ES2, 1, [ ])
+                AC_DEFINE(SDL_VIDEO_OPENGL_ES, 1, [ ])
+                AC_DEFINE(SDL_VIDEO_RENDER_OGL_ES, 1, [ ])
+                AC_DEFINE(SDL_VIDEO_RENDER_OGL_ES2, 1, [ ])
+                EXTRA_CFLAGS="$EXTRA_CFLAGS -DGLES_SILENCE_DEPRECATION"
+            fi
+            SOURCES="$SOURCES $srcdir/src/video/uikit/*.m"
+            SUMMARY_video="${SUMMARY_video} uikit"
+            have_video=yes
         fi
-        SOURCES="$SOURCES $srcdir/src/video/uikit/*.m"
-        SUMMARY_video="${SUMMARY_video} uikit"
-        have_video=yes
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lm"
         if test x$enable_system_iconv = xyes; then
             EXTRA_LDFLAGS="$EXTRA_LDFLAGS -liconv"
